### DISCUSSION TITLE
I want to use cloud servers with dedicated boxes

### DISCUSSION THREAD

**Discussion by ghost** [kube-hetzner/terraform-hcloud-kube-hetzner#517] [Ideas]
So in short, my control plane, auto scaling nodes and so on are cloud machines
While also running one or more dedicated boxes (in the same data center) for stateless workloads.
I imagine this wouldn't be much of an issue - I've already done something in this direction, but using a different k8s distro.
There is the odd manual step, on the dedicated side - setting up the networking, but that's pretty much a one off and not hard to do at all.

So... my question: would I be able to just add another metal box to the cluster, after initially setting it up?

And a sidenote - I had a loooot of issues with volumes on hetzner with k3s  and hetzner csi (which is why I stopped using it on hetzner) -have you not had issues, at all? (issues were volumes wouldn't successfully detach and re-attach to another node)

EDIT: I see there is a similar discussion in issues, where I only searched for this in discussions, so maybe close this

**Comment by mysticaltech** [comment]
@crisrat Yes, @maaft Got it working, but it's not officially supported for now, https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/433

About volumes, I did not know of significant issues with HetznerCSI, but it's completely possible indeed. We also support Longhorn, which I think is more flexible. You can check our docs for more details on this.

**Reply by ghost** [reply]
@mysticaltech I've been answering people and telling them this in the issues on that repo
I had that issue so many times I stopped using k3s once I realized it's caused by k3s / hetzner-csi combo. If you look on the hetzner-csi repo, you'll see a lot of discussions.
Yeah, longhorn is really nice as an alternative :)
I presume if longhorn is enabled, then volumes are not used

**Reply by mysticaltech** [reply]
Good to hear @crisrat! About Longhorn with volumes, we also support that. Please see the docs how to enable that!

**Reply by ghost** [reply]
cheers
Not sure I am brave enough to try volumes with k3s again :)

**Reply by maaft** [reply]
I've got no issues with k3s + hetzner-csi (except in my cloud/bare-metal cluster)

**Reply by ghost** [reply]
maybe they fixed it - good to hear

**Reply by maaft** [reply]
When did you have these issues?

Disclaimer: I'm only running the stack in a staging env since roughly 2 weeks. I've got a vault cluster and a postgres cluster. Both did not had any volume issues during node upgrades. I'm not 100% sure, if the node where the pods ran ever changed though.

**Reply by ghost** [reply]
@maaft I had the issues a few months ago, less than a year
Initial create and attach worked fine, but if I deleted a note (for example when testing auto scaling etc), it would fail to re-attach the volume to a new virtual machine

**Reply by mysticaltech** [reply]
@crisrat The failure to reattach sounds familiar lol. But there are ways to make it work if I remember correctly! Maybe searching this project will give you solutions.

**Reply by ghost** [reply]
@mysticaltech I tried lots of things, and searched high and low - on the hetzner-csi repo especially, but with mixed success
For me, if it doesn't work without my intervention, it is not good enough, so I just stopped using k3s on hetzner

**Reply by mysticaltech** [reply]
@crisrat It's not k3s per se, it's probably just the config of the CSI. You should not have this problem with Longhorn!

**Reply by ghost** [reply]
@mysticaltech not sure, I've since switched to kubeone and that has no issues at all with mounting volumes - regularly auto scales, I've deleted all nodes just to see how it scales back up and all volumes, always mount. perhaps with the new csi version they released, I think it's at 2 now(?), issues are resolved

**Reply by mysticaltech** [reply]
@crisrat Interesting, good to know!
